internal/controller/eventing/integrationtests/validation/integration_test.go

```diff
@@ -0,0 +1,683 @@
+							logLevel: gibberish,
+						},
+					},
+				},
+			},
+			wantErrMsg: "logLevel can only be set to Debug, Info, Warn or Error",
+		},
+		{
+			name: `validation of spec.logging passes for value = Debug`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelDebug,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Info`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelInfo,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Warn`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelWarn,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Error`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelError,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			testEnvironment.EnsureNamespaceCreation(t, tc.givenUnstructuredEventing.GetNamespace())
+
+			// when
+			err := testEnvironment.CreateUnstructuredK8sResource(&tc.givenUnstructuredEventing)
+
+			// then
+			if tc.wantErrMsg == noError {
+				require.NoError(t, err, "Expected error message to be empty but got error instead."+
+					" Check the validation rule of the eventing CR.")
+			} else {
+				require.Error(t, err, fmt.Sprintf("Expected the following error message: \n \" %s \" \n"+
+					" but got no error. Check the validation rules of the eventing CR.", tc.wantErrMsg))
+
+				require.Contains(t, err.Error(), tc.wantErrMsg, "Expected a specific error message"+
+					" but messages do not match. Check the validation rules of the eventing CR.")
+			}
```

Comment:
(nit) I believe all of this code block can be simplified to require.Equal(t, wantError, gotError).

To achieve this you need to change the `tc.wantErrMsg` to `tc.wantErroor`, where `wantErroor` is of type `error`.